bpython/curtsiesfrontend/repl.py

```diff
@@ -860,6 +867,9 @@ def readline_kill(self, e):
 
     def on_enter(self, insert_into_history=True, reset_rl_history=True):
         # so the cursor isn't touching a paren TODO: necessary?
+        if insert_into_history:
```

Comment:
This if statement reads confusingly to me, I think we should:
- rename the `insert_into_history` parameter in `Repl.reevaluate()` to `new_code`; this is what it means, it's about whether the lines might be new, so they should be added to readline history.
- rename the `insert_into_history` parameter in `Repl.on_enter()` to `new_code` as well. Then this if becomes "if the line being executed is new, then we can no longer use our redo stack."
- not rename insert_into_history anywhere else

My rationale for the name is that `new_code` describes the situation/circumstance instead of the behavior we want, which is important since we're now using this flag to determine a new, different behavior.

And in reviewing this I found that the `self.reevaluate(insert_into_history=True)` in `clear_modules_and_reevaluate()` should probably be False instead, we can fix in another PR.